### PR TITLE
Bluesky Vector Scan Conversion

### PR DESCRIPTION
**Pull request by mskinner5278**
Draft for moving the parameters for the vector scan from epics PVs to the bluesky flyer
#164 also one of the components we will need for #161

### CODE REVIEW

File: daq_macros.py [hunk 1]
```diff
@@ -2563,27 +2563,33 @@ def vectorZebraScanNormal(vecRequest):
...
+  else:
+      transmission = getPvDesc("transmissionRBV")
+
+  reqObj["vectorParams"]["x_beam"]=x_beam
+  reqObj["vectorParams"]["y_beam"]=y_beam
+  reqObj["vectorParams"]["wavelength"]=wavelength
+  reqObj["vectorParams"]["det_distance_m"]=det_distance_m
+  reqObj["vectorParams"]["transmission"]=transmission
+
+  zebraDaqBluesky(flyer,swee_start_angle,range_degrees,imgWidth,expTime,file_prefix,data_directory_name,file_number_start, vector_params=reqObj["vectorParams"])
```
Comment (RobertSchaffer1) [inline daq_macros.py:2592]: Typo: swee_start -> sweep_start

**Comment by RobertSchaffer1** [comment]
Generally I think this looks good, and I've completely forgotten where I was going to suggest additional error handling... We should test pre-merge, and wait for Jun's feedback as well.

File: daq_macros.py [hunk 2]
```diff
@@ -3117,20 +3131,46 @@ def zebraDaqBluesky(flyer, angle_start, scanWidth, imgWidth, exposurePeriodPerIm
...
 
     flyer.update_parameters(angle_start=angle_start, scan_width=scanWidth, img_width=imgWidth, num_images=num_images, exposure_period_per_image=exposurePeriodPerImage, \
-                   x_start_um=current_x, y_start_um=current_y, z_start_um=current_z, \
+                   x_start_um=x_vec_start, y_start_um=y_vec_start, z_start_um=z_vec_start, \
+                   x_end_um=x_vec_end, y_end_um=y_vec_end, z_end_um=z_vec_end, \
                    file_prefix=filePrefix, data_directory_name=data_directory_name, file_number_start=file_number_start,\
-                   x_beam=x_beam, y_beam=y_beam, wavelength=wavelength, det_distance_m=det_distance_m,\
-                   detector_dead_time=0.024, scan_encoder=scanEncoder, change_state=changeState, transmission=transmission)
+                   x_beam=vector_params["x_beam"], y_beam=vector_params["y_beam"], wavelength=vector_params["wavelength"], det_distance_m=vector_params["det_distance_m"],\
+                   detector_dead_time=0.024, scan_encoder=scanEncoder, change_state=changeState, transmission=vector_params["transmission"])
```
Comment (JunAishima) [inline daq_macros.py:3172]: can you extract the dead time? this should be something like flyer.detector.dead_time.get() for Eiger detectors. that value is for Pilatus

File: daq_macros.py [hunk 3]
```diff
@@ -2563,27 +2563,33 @@ def vectorZebraScanNormal(vecRequest):
...
-  setPvDesc("vectorEndOmega",sweep_end_angle)  
-  setPvDesc("vectorStartX",x_vec_start)
-  setPvDesc("vectorStartY",y_vec_start)  
-  setPvDesc("vectorStartZ",z_vec_start)  
-  setPvDesc("vectorEndX",x_vec_end)
-  setPvDesc("vectorEndY",y_vec_end)  
-  setPvDesc("vectorEndZ",z_vec_end)  
-  setPvDesc("vectorframeExptime",expTime*1000.0)
-  setPvDesc("vectorNumFrames",numImages)
   scanWidth = float(numImages)*imgWidth
```
Comment (JunAishima) [inline daq_macros.py:2570]: one of either range_degrees or scanWidth should be used - they are the same value and used in either the call to zebraDaq() or zebraDaqBluesky()

for code similarity, I would suggest removing range_degrees - although that is more intuitive code than scanWidth

**Review by JunAishima — CHANGES_REQUESTED** [review]

File: daq_macros.py (hunk 3 above)
Comment (mskinner5278) [reply daq_macros.py:2569]: removed range_degrees in 3fbdaa8

File: daq_macros.py (hunk 2 above)
Comment (mskinner5278) [reply daq_macros.py]: extracted dead time and added handling for attribute error in 1d73c26

**Review by JunAishima — APPROVED** [review]

**Pull request merged**